### ISSUE TITLE
Various updates to SqlServerSyntax are needed

### ISSUE THREAD

**Issue by RachelAmbler** [erikbra/grate#243]
Multiple issues: Things like `next` should be `NVarChar(Max), Double quotes are used where [] should be, the `ListDatabases` property needs to exclude system database, read only databases and unavailable databases. [] should be added in multiple other places to avoid potential errors.

I've already made these changes and I'll raise a PR for these later on.

**Comment by erikbra** [comment]
Mostly solved by #501 - please open another issue if you see anything else that should be changed. It's *way* easier to update the grate structure after #482 :)
